resolwe_bio/processes/fusion_calling/arriba.py

```diff
@@ -0,0 +1,126 @@
+"""Gene fusion detection with Arriba."""
+
+import pandas as pd
+from plumbum import TEE
+
+from resolwe.process import Cmd, DataField, FileField, Process, SchedulingClass
+
+
+def get_contig_names(gtf_file):
+    """Get unique contig names.
+
+    List of contig (chromosome) names is required by arriba (parameter -i).
+    This function covers possible edge cases where contig names are not common.
+    """
+
+    gtf = pd.read_csv(gtf_file, sep="\t", header=None, usecols=[0])
+    contigs = set(gtf[0])
+    out = " ".join(list(contigs))
+    return out
+
+
+class Arriba(Process):
+    """Run Arriba for fusion detection from RNA-Seq data.
+
+    Arriba is a command-line tool for gene fusion detection from RNA-Seq data.
+    This process detects gene fusions from RNA-Seq data using the Arriba tool.
+    The input can be a BAM file from the STAR aligner, and additional optional
+    inputs such as blacklist, protein domains, and known fusions files can be provided.
+    More information about Arriba can be found in the
+    [Arriba manual](https://arriba.readthedocs.io/en/latest/) and in
+    the [original paper](https://genome.cshlp.org/content/31/3/448)
+    The current version of Arriba is 2.4.0.
+    """
+
+    slug = "arriba"
+    name = "Arriba"
+    process_type = "data:genefusions:arriba"
+    version = "1.0.0"
+    category = "Gene fusions"
+    scheduling_class = SchedulingClass.BATCH
+    entity = {"type": "sample"}
+    requirements = {
+        "expression-engine": "jinja",
+        "executor": {
+            "docker": {"image": "public.ecr.aws/genialis/resolwebio/rnaseq:6.3.0"}
+        },
+        "resources": {
+            "cores": 1,
+            "memory": 12288,
+        },
+    }
+    data_name = '{{ bam|name|default("?") }}'
+
+    class Input:
+        """Input fields for Arriba process."""
+
+        bam = DataField("alignment:bam:star", label="Input BAM file from STAR aligner")
```

Comment:
Label could go more in depth of what kind of bam file is expected (how it's processed in i.e. STAR).